R/Set_up_subfunctions.R

```diff
@@ -0,0 +1,338 @@
+#' @title
+#' Calculate key age parameters to inform life tables (for pig population)
+#' @description
+#' Calculate age rate and age widths for pig compartments depending on number of age groups 
+#'
+#' @param number_age_classes the number of age classes (pigs)
+#' @param pig_age_class_width vector for mean no. of months in each age compartment 
+#'
+#' @return age rate and age width
+#' @export
+age_parameters_pig_func <- function(number_age_classes, pig_age_class_width){
+  
+  number_age_classes_pig = number_age_classes
+  
+  # vector for mean no. of months in each age compartment 
+  
+  age_width_pig <- c()
+  
+  #=======================================#
+  # preparing age_width & age_rate params #
+  
+  # age_width calc: Age-structured model #
+  if (number_age_classes_pig > 1) {
+    for (i in number_age_classes_pig) {
+      # width of 1 month for each age class (symmetrical age classes)
+      age_width_pig[1:number_age_classes_pig] <- pig_age_class_width
+      }
+    }
+  
+  # create vector of length n age classes for ODIN (& create dimensions for other variables)
+  if (number_age_classes_pig == 1) {
+    na_pig <- 1
+    }
+  
+  if (number_age_classes_pig > 1) {
+    na_pig <- as.integer(length(age_width_pig)) 
+    }
```

Comment:
This is needed because `age_width_pig` is NULL when `number_age_classes_pig`=1, so if  only have the argument `na_pig <- as.integer(length(age_width_pig))` then `na_pig `= 0 (rather than explicit logic to specify `na_pig` = 1)